### PR TITLE
ci: refine template packing process

### PR DESCRIPTION
**Pull request by hund030**
1. Remove fx-core’s pre-build script. 
1. Remove generate-template-zip.sh script which is used in CD action. 
1. Add build script for templates, which generate zip files and distribute them to fx-core’s fallback folder and temporary folder in CD action for release. 
1. Set useLocalTemplate as true by default.

### CODE REVIEW

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -134,14 +132,12 @@ jobs:
...
 
-      - name: generate templates v3
+      - name: rebuild templates after versioning
         run: |
-          .github/scripts/template-zip-autogen-v3.sh $TEMPLATE_PATH
-          cd ./packages/fx-core
-          rm -rf ./templates/fallback
-          node ./scripts/download-templates-zip.js
+          cd ./templates
+          npm run build
```
Comment (wenytang-ms) [inline .github/workflows/cd.yml:138]: why need rebuild?

Comment (hund030) [reply .github/workflows/cd.yml:138]: Templates' package.json may be changed after versioning, which may sync up sdk version.

Comment (wenytang-ms) [reply .github/workflows/cd.yml:138]: what if do these thing in postversion?

Comment (hund030) [reply .github/workflows/cd.yml]: Good suggestion. I have updated.

File: .github/scripts/fxcore-sync-up-version.js [hunk 2]
```diff
@@ -29,32 +29,36 @@ if (!semver.prerelease(templateVersion)) {
...
   fse.writeFileSync(
     templateConfig,
     JSON.stringify(templateConfigFile, null, 4)
   );
 } else if (templateVersion.includes("alpha")) {
   console.log("sync up template in fx-core as 0.0.0-alpha");
   templateConfigFile.version = "0.0.0-alpha";
-  templateConfigFile.tagPrefix = "templates-";
   templateConfigFile.useLocalTemplate = true;
+  templateConfigFile.tagPrefix = "templates-";
```
Comment (wenytang-ms) [inline .github/scripts/fxcore-sync-up-version.js:52]: still need alpha, beta and others ?

**Comment by hund030** [comment]
Test results:
https://github.com/hund030/TeamsFx/actions/runs/6532163201

File: .github/scripts/fxcore-sync-up-version.js (hunk 2 above)
Comment (hund030) [reply .github/scripts/fxcore-sync-up-version.js]: removed

**Review by wenytang-ms — APPROVED** [review]

**Review by eriolchan — APPROVED** [review]

**Review by huimiu — APPROVED** [review]

**Pull request merged**